src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.cpp

```diff
@@ -1665,11 +1676,21 @@ void Partitioner::optimize(const std::string& func_name) {
             }
             ov::parallel_for(func_group.refs.size(), [&](std::size_t f_idx) {
                 auto& funcall = func_group.refs[f_idx].get();
-                std::vector<ov::Tensor> to_concat;
+                std::vector<LazyTensor> to_concat;
+                // Fill tensor vector
                 for (auto&& cidx : to_concat_idx) {
-                    to_concat.push_back(funcall._closure[cidx]);
+                    // FIXME: Assuming here concat goes first and other transformations later.
+                    //        This allows to store ov::Tensor and ignore their potential history of transformations
+                    NPUW_ASSERT(!funcall._lazy_closure[cidx].has_transformations());
+                    to_concat.push_back(funcall._lazy_closure[cidx]);
+                }
+                // Note: we can ignore updating funcall._lazy_closure[cidx] here since those LazyTensors will be gone
+                // and the new one added into the vector
+                if (!to_concat.empty()) {
+                    funcall._lazy_closure.push_back(LazyTensor(TransformType::CONCAT, std::make_pair(to_concat, axis)));
+                    // Some of the tensors might be in closure - preserve it's 1:1 idx mapping with _lazy_closure
+                    funcall._closure.resize(funcall._lazy_closure.size());
```

Comment:
I think, for consistency it could be just a plain `.push_back(ov::Tensor())` -- `.push_back()` here, `.push_back()` there
```suggestion
                    funcall._closure.push_back(ov::Tensor());
```

Reply:
Done